pkg/asset/installconfig/vsphere/validation.go

```diff
@@ -50,19 +83,31 @@ func folderExists(ic *types.InstallConfig, fldPath *field.Path) field.ErrorList
 		return allErrs
 	}
 
-	vim25Client, _, err := vspheretypes.CreateVSphereClients(context.TODO(), cfg.VCenter, cfg.Username, cfg.Password)
-	if err != nil {
-		err = errors.Wrap(err, "unable to connect to vCenter API")
-		return append(allErrs, field.InternalError(fldPath, err))
-	}
+	ctx, cancel := context.WithTimeout(context.TODO(), 60*time.Second)
+	defer cancel()
 
-	finder := find.NewFinder(vim25Client)
+	if _, err := finder.Folder(ctx, cfg.Folder); err != nil {
+		return append(allErrs, field.Invalid(fldPath, cfg.Folder, err.Error()))
+	}
+	return nil
+}
 
+func validateNetwork(finder VSphereClient, p *vsphere.Platform, fldPath *field.Path) field.ErrorList {
 	ctx, cancel := context.WithTimeout(context.TODO(), 60*time.Second)
 	defer cancel()
+	dcName := p.Datacenter
+	if !strings.HasPrefix(dcName, "/") && !strings.HasPrefix(dcName, "./") {
+		dcName = "./" + dcName
+	}
 
-	if _, err = finder.Folder(ctx, cfg.Folder); err != nil {
-		return append(allErrs, field.Invalid(fldPath, cfg.Folder, err.Error()))
+	dataCenter, err := finder.Datacenter(ctx, dcName)
+	if err != nil {
+		return field.ErrorList{field.Invalid(fldPath, p.Datacenter, "unable to find datacenter provided")}
```

Comment:
Should we include `err` in the field error?
```suggestion
		return field.ErrorList{field.Invalid(fldPath, p.Datacenter, fmt.Sprintf("unable to find datacenter provided: %v", err))}
```
Or, should we keep consistent with the field error when we cannot find the folder?
```suggestion
		return field.ErrorList{field.Invalid(fldPath, p.Datacenter, err.Error())}
```

Reply:
I'll change it to the latter for consistency. Let me know if the former is better.